### PR TITLE
Revise instruction on running local Typesense instance

### PR DESCRIPTION
**Pull request by kianmeng**
We switch to Docker Compose for local Typesense setup and improves the documentation structure and clarity.

### CODE REVIEW

**Comment by kianmeng** [comment]
@jaeyson Perhaps we should remove the TODO, move to GitHub issues, and add link to readme instead?

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+services:
+  typesense:
+    stdin_open: true
+    tty: true
+    container_name: typesense
+    environment:
+      - TYPESENSE_DATA_DIR=/data
+      - TYPESENSE_API_KEY=xyz
+    volumes:
+      - ./data:/data
```
Comment (kianmeng) [inline docker-compose.yml:11]: No need to manually create the data folder, as in `mkdir /tmp/typesense-server-data`.

**Comment by jaeyson** [comment]
> @jaeyson Perhaps we should remove the TODO, move to GitHub issues, and add link to readme instead?

Good idea! :+1:

**Review by jaeyson — APPROVED** [review]

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Pull request merged**